store/index/index.go

```diff
@@ -378,7 +378,10 @@ func (i *Index) Put(key []byte, location types.Block) error {
 				return nil
 			}
 
-			trimmedPrevKey := prevKey[:keyTrimPos+1]
+			trimmedPrevKey := prevKey
+			if keyTrimPos < len(prevKey) {
+				trimmedPrevKey = prevKey[:keyTrimPos+1]
+			}
```

Comment:
If this is caused by a legitimate key value, then there needs to be a unit test with that key.  If this is to get around some form of corruption, then an error should be logged in an `else` case above.  If this is a corruption work-around then a similar check should be done at line 374:
```go
if len(fullPrevKey) <= i.sizeBits / 8 {
    return fmt.Errorf("previous key too short: %d" , len(fullPrevKey))
}
```

Reply:
this is still happening regularly currently - what other logs will be useful in figuring out how to mitigate?

Reply:
It would probably make sense to log output from `readBucketInfo`:
```suggestion
			} else {
				cached, indexOffset, recordListSize, fileNum, err := i.readBucketInfo(bucket)
				if err != nil {
					log.Errorw("Cannot read bucket", "err", err)
				} else {
					log.Errorw("Read bad pevious key data", "cached", cached, "pos", indexOffset, "size", recordListSize, "file", indexFileName(i.basePath, fileNum))
				}
			}
```